server/resourcemanager/testutil/operations.go

```diff
@@ -1,115 +1,46 @@
 package testutil
 
 import (
-	"encoding/json"
-	"io"
 	"net/http"
+	"net/http/httptest"
 	"testing"
-
-	"github.com/stretchr/testify/require"
 )
 
+type Operation string
+
+type operationTester struct {
+	buildRequest   func(*testing.T, *httptest.Server, contentTypeConverter, ResourceTypeTest) *http.Request
+	assertResponse func(*testing.T, *http.Response, contentTypeConverter, ResourceTypeTest)
+	name           Operation
+	postAssert     func(t *testing.T, ct contentTypeConverter, rt ResourceTypeTest, testServer *httptest.Server)
+}
+
 var (
 	defaultOperations = []operationTester{
-		createNoIDOperation{},
-		createSuccessOperation{},
+		createNoIDOperation,
+		createSuccessOperation,
 
-		updateNotFoundOperation{},
-		updateSuccessOperation{},
+		updateNotFoundOperation,
+		updateSuccessOperation,
 
-		getNotFoundOperation{},
-		getSuccessOperation{},
+		getNotFoundOperation,
+		getSuccessOperation,
 
-		deleteNotFoundOperation{},
-		deleteSuccessOperation{},
+		deleteNotFoundOperation,
+		deleteSuccessOperation,
 
-		listNoResultsOperation{},
-		listSuccessOperation{},
+		listNoResultsOperation,
+		listSuccessOperation,
+		listPaginatedAscendingSuccessOperation,
+		listPaginatedDescendingSuccessOperation,
 		// TODO: add tests for pagination etc
 	}
 
 	errorOperations = []operationTester{
-		createInteralErrorOperation{},
-		updateInteralErrorOperation{},
-		getInteralErrorOperation{},
-		deleteInteralErrorOperation{},
-		listInteralErrorOperation{},
+		createInternalErrorOperation,
+		updateInternalErrorOperation,
+		getInternalErrorOperation,
+		deleteInternalErrorOperation,
+		listInternalErrorOperation,
 	}
 )
-
-func removeID(input map[string]any) map[string]any {
```

Comment:
As these functions are helpers, not related directly to operations, I've moved them to`common_functions.go`.